src/main/java/com/example/usermanagement/implementation/UserService.java

```diff
@@ -23,32 +26,43 @@ public class UserService {
     @Autowired
     UserExisting userExisting;
 
-    public Response create(CreateUserdto createUserdto, List<User> userList) throws UserAlreadyExistsException {
-        for (User user : userList) {
-            if (user.getUsername().equals(createUserdto.getUsername())) {
-                throw new UserAlreadyExistsException(String.format(UserConstants.USER_ALREADY_EXISTS, createUserdto.getUsername()));
-            }
+    @Autowired
+    UserRepository userRepository;
+
+    public Response create(CreateUserdto createUserdto) throws UserAlreadyExistsException {
+        String username = createUserdto.getUsername();
+        User user = userRepository.findByUsername(username);
+        if (user != null) {
+            throw new UserAlreadyExistsException(String.format(UserConstants.USER_ALREADY_EXISTS, username));
         }
-        return userMapper.createMapper(createUserdto, userList);
+        user = userMapper.mapToUser(createUserdto);
+        userRepository.save(user);
+        createUserdto.setId(user.getId());
+        return userMapper.createMapper(user);
     }
 
-    public Response update(UpdateUserDto updateUserDto, List<User> userList, String username) throws UserNotExistException {
-        userExisting.ifUserExist(userList, username);
-        return userMapper.updateMapper(updateUserDto, userList, username);
+    public Response update(UpdateUserDto updateUserDto, Long userId) throws UserNotExistException {
+        userExisting.ifUserExist(userId);
+        Optional<User> user = userRepository.findById(userId);
```

Comment:
You can return the found user from this function instead of performing findById again